src/mpid/ch4/src/ch4_types.h

```diff
@@ -282,7 +282,7 @@ typedef struct MPIDI_CH4_Global_t {
 
     int n_vcis;
     MPIDI_vci_t vci[MPIDI_CH4_MAX_VCIS];
-    int progress_counts[MPIDI_CH4_MAX_VCIS];
+    MPL_atomic_int_t progress_counts[MPIDI_CH4_MAX_VCIS];
```

Comment:
The reason this was not atomic is we nearly always update and access progress counts inside a per-vci critical section -- except for progress wait. With progress, we already use atomics for completion counters inside the requests, so it appears that we should have better optimization that avoids the use of the progress counter altogether. 

Anyway, I think using relaxed atomics should be fine. Could you add a comment here explain that the atomics is only needed by progress_wait, and we only use relaxed semantics for it?

Reply:
Thanks. I added a comment there.